neuro_san/session/http_service_agent_session.py

```diff
@@ -13,119 +13,20 @@
 from typing import Any
 from typing import Dict
 from typing import Generator
+
 import json
 import requests
 
-
-from leaf_common.time.timeout import Timeout
-
 from neuro_san.interfaces.agent_session import AgentSession
+from neuro_san.session.abstract_http_service_agent_session import AbstractHttpServiceAgentSession
 
 
-class HttpServiceAgentSession(AgentSession):
+class HttpServiceAgentSession(AbstractHttpServiceAgentSession, AgentSession):
     """
     Implementation of AgentSession that talks to an HTTP service.
     This is largely only used by command-line tests.
     """
 
-    DEFAULT_AGENT_NAME: str = "esp_decision_assistant"
-
-    # pylint: disable=too-many-arguments,too-many-positional-arguments
-    def __init__(self, host: str = None,
-                 port: str = None,
-                 timeout_in_seconds: int = 30,
-                 metadata: Dict[str, str] = None,
-                 security_cfg: Dict[str, Any] = None,
-                 umbrella_timeout: Timeout = None,
-                 streaming_timeout_in_seconds: int = None,
-                 agent_name: str = DEFAULT_AGENT_NAME):
-        """
```

Comment:
All these methods move to AbstractHttpServiceAgentSessions to have commonality with async version.